manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java

```diff
@@ -187,20 +187,28 @@ public Processor updateProcessor(String bridgeId, String processorId, String cus
         if (!Objects.equals(existingProcessor.getType(), processorRequest.getType())) {
             throw new BadRequestException("It is not possible to update the Processor's Type.");
         }
-        // See https://issues.redhat.com/browse/MGDOBR-516 for updating Action support
-        if (!Objects.equals(existingAction, processorRequest.getAction())) {
-            throw new BadRequestException("It is not possible to update the Processor's Action.");
+        if (existingProcessor.getType() == ProcessorType.SINK) {
+            if (!Objects.equals(existingAction.getType(), processorRequest.getAction().getType())) {
+                throw new BadRequestException("It is not possible to update the Processor's Action Type.");
+            }
         }
-        if (!Objects.equals(existingSource, processorRequest.getSource())) {
-            throw new BadRequestException("It is not possible to update the Processor's Source.");
+        if (existingProcessor.getType() == ProcessorType.SOURCE) {
+            if (!Objects.equals(existingSource.getType(), processorRequest.getSource().getType())) {
+                throw new BadRequestException("It is not possible to update the Processor's Source Type.");
+            }
         }
 
         // Construct updated definition
         Set<BaseFilter> updatedFilters = processorRequest.getFilters();
         String updatedTransformationTemplate = processorRequest.getTransformationTemplate();
+        Source updatedSource = processorRequest.getSource();
+        Action updatedAction = processorRequest.getAction();
+        Action updatedResolvedAction = processorRequest.getType() == ProcessorType.SOURCE
+                ? resolveSource(processorRequest.getSource(), customerId, bridgeId, processorId)
+                : resolveAction(processorRequest.getAction(), customerId, bridgeId, processorId);
         ProcessorDefinition updatedDefinition = existingProcessor.getType() == ProcessorType.SOURCE
-                ? new ProcessorDefinition(updatedFilters, updatedTransformationTemplate, existingSource, existingResolvedAction)
-                : new ProcessorDefinition(updatedFilters, updatedTransformationTemplate, existingAction, existingResolvedAction);
+                ? new ProcessorDefinition(updatedFilters, updatedTransformationTemplate, updatedSource, updatedResolvedAction)
+                : new ProcessorDefinition(updatedFilters, updatedTransformationTemplate, updatedAction, updatedResolvedAction);
 
         // No need to update CRD if the definition is unchanged
         if (existingDefinition.equals(updatedDefinition)) {
```

Comment:
No, since this check also considers Filters and Transformations. 

However I will need to refactor it given @sutaakar comments about secret parameters too.

Reply:
@robpblake This is a short-circuit to prevent an unnecessary CRD provisioning cycle if the definition had not changed.

However now we have the approaching issue of _secret_ parameters this check can no longer work.

I don't think we have much of an option other than to remove it.

WDYT?